test/integrators/step_limiter_test.jl

```diff
@@ -0,0 +1,57 @@
+using OrdinaryDiffEq, Test
+
+# define the counting variable
+const STEP_LIMITER_VAR = Ref(0)
+# define the step_limiter! function which just counts the number of step_limiter calls
+step_limiter!(u, integrator, p, t) = STEP_LIMITER_VAR[] += 1
+
+
+# This function tests the step limiter functionality of an ODE solver.
+function test_step_limiter(alg_type, adaptive = true)
+    STEP_LIMITER_VAR[] = 0 # reset the counting variable
+    prob = ODEProblem((du, u, p, t) -> du .= u, [1.0], (0.0, 5.0))
+
+    if adaptive == true
+        sol = solve(prob, alg_type(; step_limiter!))
+    else
+        sol = solve(prob, alg_type(; step_limiter!), dt = 1)
+    end
+
+    n = sol.stats.naccept + sol.stats.nreject
+    @test n == STEP_LIMITER_VAR[]
+end
+
+
+@testset "Step_limiter Test" begin
+    # it only catches the most basic errors, i.e. if the step_limiter! function is not called
+    # or called more then one time
+
+    # I think all the algorithms with step_limiter! function are tested here
+    # (besides Low Storage Runge-Kutta methods)
+
+    STEP_LIMITER_VAR[] = 0
```

Comment:
```suggestion
```
No need for this here since you reset the counter in `test_step_limiter`, don't you?

Reply:
It shouldn't be needed. I just wanted to be safe